llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1255,7 +1255,8 @@ void AddressSanitizer::instrumentMemIntrinsic(MemIntrinsic *MI) {
   InstrumentationIRBuilder IRB(MI);
   if (isa<MemTransferInst>(MI)) {
     IRB.CreateCall(isa<MemMoveInst>(MI) ? AsanMemmove : AsanMemcpy,
-                   {MI->getOperand(0), MI->getOperand(1),
+                   {IRB.CreateAddrSpaceCast(MI->getOperand(0), PtrTy),
+                    IRB.CreateAddrSpaceCast(MI->getOperand(1), PtrTy),
```

Comment:
do we need it in memset?

Reply:
I don't think we need currently this for memset.

Reply:
It will be needed for any memory intrinsic

Reply:
I guess never-`const` applies to the first arg of memcpy, but the patch adds CreateAddrSpaceCast?
So maybe use CreateAddrSpaceCast for all intrinsics for consistency

Reply:
> > It will be needed for any memory intrinsic
> 
> Logically we should never see an attempt to set memory with a pointer to constant memory, but I think we need to handle memset otherwise.

The "constant" part here isn't what matters, it's the non-0 address space

Reply:
Hi all,

The latest PR is https://github.com/llvm/llvm-project/pull/79795 to fix this issue.
Thanks again for all of your inputs.